recipes-bsp/fpga/fpga-image-download.inc

```diff
@@ -0,0 +1,20 @@
+# Download binary bitstream instead of compiling it
+SUMMARY = "FPGA reference image"
+require fpga-image.inc
+LICENSE = "CLOSED"
+
+# Expect to get PV and checksums from bb file that includes this file
+
+TOPICDOWNLOADS_URI ?= "http://topic-downloads.fra1.digitaloceanspaces.com"
+PACKAGENAME ?= "${BPN}-${MACHINE}-${PV}"
+BOARD_DESIGN_URI = "http://fpga-server.local/downloads/fpga/${PACKAGENAME}.bit.xz;name=${MACHINE}"
+
+PKGV = "${PV}"
+S = "${WORKDIR}"
+B = "${S}"
+
+# Nothing to build
+do_compile() {
```

Comment:
Why is there a do_compile() in the *.inc as well?

Reply:
By default it expects a plain bitstream. That's the most common case.
This particular version uses the bitstream in the xsa file (not very common) and hence overrides this.